packages/teleterm/src/ui/Identity/Identity.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useRef, useState } from 'react';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+import { Box, Flex, Text } from 'design';
+import Popover from 'design/Popover';
+import { ClusterItem } from 'teleterm/ui/Identity/ClusterItem';
+import styled from 'styled-components';
+import { ListItem } from 'teleterm/ui/Navigator/NavItem';
+import { Cluster } from 'teleterm/services/tshd/types';
+import { SortDesc } from 'design/Icon';
+
+export function Identity() {
+  const shortInfoRef = useRef<HTMLDivElement>();
+  const [isLongInfoOpened, setIsLongInfoOpened] = useState(false);
+  const ctx = useAppContext();
+
+  ctx.clustersService.useState();
+  ctx.workspacesService.useState();
+
+  function tryToChangeContext(clusterUri: string): void {
+    ctx.workspacesService.setActiveWorkspace(clusterUri);
+  }
+
+  function getActiveCluster(): Cluster | undefined {
+    const clusterUri = ctx.workspacesService.state.rootClusterUri;
+    if (!clusterUri) {
+      return;
+    }
+    return ctx.clustersService.findCluster(clusterUri);
+  }
```

Comment:
In this case I find it more readable to create a small function rather than putting that logic directly into a component. It would look more or less like it: 
```typescript
 const activeCluster =
    ctx.workspacesService.getRootClusterUri() &&
    ctx.clustersService.findCluster(ctx.workspacesService.getRootClusterUri());
```